osmoutils/binary_search_test.go

```diff
@@ -77,8 +86,8 @@ func TestBinarySearchBigDec(t *testing.T) {
 	cubicF := func(a osmomath.BigDec) (osmomath.BigDec, error) {
 		// these precision shifts are done implicitly in the int binary search tests
 		// we keep them here to maintain parity between test cases across implementations
-		calculation := a.Quo(osmomath.NewBigDec(10).Power(18))
-		result := calculation.Power(3)
+		calculation := a.Power(3)
+		result := calculation.Quo(osmomath.NewBigDec(10).Power(18))
 		output := result.Mul(osmomath.NewBigDec(10).Power(18))
```

Comment:
Why do we flip these two here? 1. The original ordering was intentional to have the tests aligned with our regular binary search tests and 2. the final version of this divides by 10^18 then multiplies by it right after

I suggest we revert this to maintain parity across binary search tests

Reply:
oh, I left a comment under an initial issue https://github.com/osmosis-labs/osmosis/issues/2807#issuecomment-1260660342. Long story short this function seemed not to work correctly, so I decided to do some swaps and it started working fine afterwards. More details about it are in that comment. I really hope it will clarify my decision!

PS: my bad I forgot to mention that cubicF also seemed to work incorrectly in TestBinarySearch as well, for example:

```
DEBUG:  1024195081273344 (function input) -> 0.001024195081273344 (dec version) -> 1074355614 (output)
```